File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/statistics/Statistics.java
package coolpharaoh.tee.speicher.tea.timer.views.statistics;

import android.app.AlertDialog;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import java.util.Objects;

import coolpharaoh.tee.speicher.tea.timer.R;

// This class has 9 Parent because of AppCompatActivity
@SuppressWarnings("java:S110")
public class Statistics extends AppCompatActivity {

    private HorizontalBarGraph horizontalBarGraph;
    private StatisticsViewModel statisticsViewModel;
    private int checkedItem = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_statistics);
        defineToolbarAsActionbar();
        enableAndShowBackButton();

        statisticsViewModel = new StatisticsViewModel(getApplication());

        horizontalBarGraph = new HorizontalBarGraph(findViewById(R.id.horizontal_graph_statistics), this);
        horizontalBarGraph.display(statisticsViewModel.getStatisticsOverall());
    }

    private void defineToolbarAsActionbar() {
        final Toolbar toolbar = findViewById(R.id.tool_bar);
        final TextView mToolbarCustomTitle = findViewById(R.id.tool_bar_title);
        mToolbarCustomTitle.setText(R.string.statistics_heading);
        setSupportActionBar(toolbar);
        Objects.requireNonNull(getSupportActionBar()).setTitle(null);
    }

    private void enableAndShowBackButton() {
        Objects.requireNonNull(getSupportActionBar()).setHomeButtonEnabled(true);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public boolean onCreateOptionsMenu(final Menu menu) {
        final MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_statistics, menu);

        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(final MenuItem item) {
        int id = item.getItemId();

        if (id == R.id.action_statistics_period) {
            dialogSortOption();
        }

        return super.onOptionsItemSelected(item);
    }

    private void dialogSortOption() {
        final String[] items = getResources().getStringArray(R.array.statistics_category);

        // Creating and Building the Dialog
        final AlertDialog.Builder builder = new AlertDialog.Builder(this, R.style.dialog_theme);
        builder.setIcon(R.drawable.statistics_black);
        builder.setTitle(R.string.statistics_dialog_title);
        builder.setSingleChoiceItems(items, checkedItem, (dialog, item) -> {
            checkedItem = item;
            switch (checkedItem) {
                case 0:
                    horizontalBarGraph.display(statisticsViewModel.getStatisticsOverall());
                    break;
                case 1:
                    horizontalBarGraph.display(statisticsViewModel.getStatisticsMonth());
                    break;
                case 2:
                    horizontalBarGraph.display(statisticsViewModel.getStatisticsWeek());
                    break;
                case 3:
                    horizontalBarGraph.display(statisticsViewModel.getStatisticsDay());
                    break;
                default:
            }
            dialog.dismiss();
        });
        builder.setNegativeButton(R.string.statistics_dialog_negative, null);
        builder.create().show();
    }
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/statistics/Statistics.kt
package coolpharaoh.tee.speicher.tea.timer.views.statistics

import android.app.AlertDialog
import android.content.DialogInterface
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import coolpharaoh.tee.speicher.tea.timer.R
import java.util.Objects

// This class has 9 Parent because of AppCompatActivity
class Statistics : AppCompatActivity() {
    private var horizontalBarGraph: HorizontalBarGraph? = null
    private var statisticsViewModel: StatisticsViewModel? = null
    private var checkedItem = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_statistics)
        defineToolbarAsActionbar()
        enableAndShowBackButton()
        statisticsViewModel = StatisticsViewModel(application)
        horizontalBarGraph =
            HorizontalBarGraph(findViewById(R.id.horizontal_graph_statistics), this)
        horizontalBarGraph!!.display(statisticsViewModel!!.statisticsWeek)
    }

    private fun defineToolbarAsActionbar() {
        val toolbar = findViewById<Toolbar>(R.id.tool_bar)
        val mToolbarCustomTitle = findViewById<TextView>(R.id.tool_bar_title)
        mToolbarCustomTitle.setText(R.string.statistics_heading)
        setSupportActionBar(toolbar)
        Objects.requireNonNull(supportActionBar)?.title = null
    }

    private fun enableAndShowBackButton() {
        Objects.requireNonNull(supportActionBar)?.setHomeButtonEnabled(true)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.menu_statistics, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        if (id == R.id.action_statistics_period) {
            dialogSortOption()
        }
        return super.onOptionsItemSelected(item)
    }

    private fun dialogSortOption() {
        val items = resources.getStringArray(R.array.statistics_category)

        // Creating and Building the Dialog
        val builder = AlertDialog.Builder(this, R.style.dialog_theme)
        builder.setIcon(R.drawable.statistics_black)
        builder.setTitle(R.string.statistics_dialog_title)
        builder.setSingleChoiceItems(items, checkedItem) { dialog: DialogInterface, item: Int ->
            checkedItem = item
            when (checkedItem) {
                0 -> {
                    horizontalBarGraph!!.reset()
                    horizontalBarGraph!!.display(statisticsViewModel!!.statisticsWeek)
                }

                1 -> {
                    horizontalBarGraph!!.reset()
                    horizontalBarGraph!!.display(statisticsViewModel!!.statisticsMonth)
                }

                2 -> {
                    horizontalBarGraph!!.reset()
                    horizontalBarGraph!!.display(statisticsViewModel!!.statisticsYear)
                }

                3 -> {
                    horizontalBarGraph!!.reset()
                    horizontalBarGraph!!.display(statisticsViewModel!!.statisticsOverall)
                }

                else -> {}
            }
            dialog.dismiss()
        }
        builder.setNegativeButton(R.string.statistics_dialog_negative, null)
        builder.create().show()
    }
}